manifests/07-downloads-deployment.yaml

```diff
@@ -89,11 +89,18 @@ spec:
               os.mkdir(arch)
               for operating_system in ['linux', 'mac', 'windows']:
                   os.mkdir(os.path.join(arch, operating_system))
+          for arch in ['arm64', 'ppc64le', 's390x']:
```

Comment:
To make this usable you would also need to update the the generated downloads links:
https://github.com/openshift/console-operator/blob/master/pkg/console/controllers/clidownloads/controller.go#L155
This will update the links in the console itself.

Reply:
This is when it would be ideal to have 1 place for this data.  

I'm tempted to suggest we would add this annotation to the deployment:

```yaml
  annotations:
    release.openshift.io/create-only: "true"
```
And then cut the `spec` and add from within the operator itself.  This would let the operator generate both the `cliDownloadsLinks` CR & this script from the same source struct.

@spadgett @wking not sure if there are any objections to further work in this deployment.  Personally I'm not super jazzed about adding logic for string manipulation, since this script is just a bit embedded string, but I don't really love the manual sync of multiple resources either.

Reply:
If the script is static, I'd keep it as a manifest. I'd only generate the downloads deployment in the operator if we need some logic, for instance, if we decide to make the node selector customizable.

cc @bparees

Reply:
Its probably time to add an `e2e` test to this and validate that an `http` request to each of these endpoints is valid.

Reply:
For docs purposes, the old link to the mirror is gone:
https://github.com/openshift/console/pull/2951/files#diff-f2e8b14c70b3c8bf99b5eed3b019db60L46
We no longer hard-code this:
https://github.com/openshift/console/pull/2951/files#diff-20ab917f3e7dc4814ce688ea3e52044eL67
Since we now do what I referenced above with the `CLIDownloadsController`.

Reply:
With openshift/oc#168's uniform paths, I think we want to lean on the consistency with something like:

```python
for arch in ['amd64', 'arm64', 'ppc64le', 's390x']:
[   os.mkdir(arch)
    for operating_system in ['linux', 'darwin', 'windows']:
        os_path = os.path.join(arch, operating_system)  # might want to translate darwin -> mac here, or we can just link to darwin paths from the console
        os.mkdir(os_path)
        basename = 'oc'
        if operating_system == 'windows':
            basename += '.exe'
        target_path = os.path.join(os_path, basename)
        source_path = os.path.join('/usr', 'share', 'openshift', '{}_{}'.format(operating_system, arch), executable)
        try:
            os.stat(source_path)
        except OSError as error:
            print(error)
            continue
        os.symlink(source_path, target_path)
        ...

Reply:
Mac and Windows still only cover one architecture, but that treats them as if they might cover all of them with an error if it (not surprisingly) does not.  That seems like a lot of overhead for a result for which we already know the answer.

Reply:
> Mac and Windows still only cover one architecture, but that treats them as if they might cover all of them with an error if it (not surprisingly) does not.

A warning, not an error.

> That seems like a lot of overhead for a result for which we already know the answer.

We don't know the answer, it depends on which cli-artifacts gets used for this deployment.  Maybe later they do grow additional architectures, or maybe we drop an arch, or grow a new OS, or whatever.  I don't think this deployment should be opinionated.  It should just look in the places that *might* be there, serve what it has, and 404 requests for content it doesn't have.

Reply:
Is it really possible to have a cli-artifacts out of sync with console-operator?  if not, then we *do* know which architectures are available, do we not?

Reply:
> Is it really possible to have a cli-artifacts out of sync with console-operator?

Sure.  For example, we could decide we want to publish a `windows/arm64` `oc`, add it to cli-artifacts, and my standard-path approach (which I just pushed up as #354), would immediately serve it (although you'd still need a console bump to get an entry in `PlatformBasedOCConsoleCLIDownloads`).  And if we wanted to avoid a `PlatformBasedOCConsoleCLIDownloads` bump, we could serve the choices dynamically from the downloads pod.  Hmm... :)

Reply:
It would definitely be nice to have a single source of truth and not have to sync these two items.  That said, most who use the console will probably want one of the "mainstream" versions of `oc`, so clusttering up the UI with all the things may not be that valuable.